File: address-geocode-canada-python-examples/get_geo_location_soap_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../address-geocode-canada-python/SOAP"))

from get_geo_location_soap import GetGeoLocationSoap


def get_geo_location_soap_sdk_go(is_live: bool, license_key: str) -> None:
    print("\r\n----------------------------------------------------")
    print("Address Geocode - Canada - GetGeoLocation - SOAP SDK")
    print("----------------------------------------------------")

    address = "1053 Carling Ave"
    municipality = "Ottawa"
    province = "ON"
    postal_code = "N0A 1E0"
    timeout_seconds = 15

    print("\r\n* Input *\r\n")
    print(f"Address        : {address}")
    print(f"Municipality   : {municipality}")
    print(f"Province       : {province}")
    print(f"PostalCode     : {postal_code}")
    print(f"License Key    : {license_key}")
    print(f"Is Live        : {is_live}")
    print(f"Timeout Seconds: {timeout_seconds}")

    try:
        service = GetGeoLocationSoap(license_key, is_live, timeout_seconds)
        response = service.get_geo_location(address, municipality, province, postal_code)

        if not hasattr(response, "Error") or not response.Error:
            print("\r\n* Geocode Info *\r\n")
            print(f"Latitude  : {response.Latitude}")
            print(f"Longitude : {response.Longitude}")
            print(f"PostalCode: {response.PostalCode}")
            print(f"MatchCode : {response.MatchCode}")
        else:
            print("No geocode info found.")

        if hasattr(response, "Error") and response.Error:
            print("\r\n* Error *\r\n")
            print(f"Error Desc    : {response.Error.Desc}")
            print(f"Error Number  : {response.Error.Number}")
            print(f"Error Location: {response.Error.Location}")

    except Exception as e:
        print("\r\n* Error *\r\n")
        print(f"Exception occurred: {str(e)}")

File: address-geocode-canada-nodejs-examples/get_geo_location_soap_sdk_example.js
import { GetGeoLocationSoap } from "../address-geocode-canada-nodejs/SOAP/get_geo_location_soap.js";

async function GetGeoLocationSoapGo(licenseKey, isLive) {
    console.log("\n----------------------------------------------------");
    console.log("Address Geocode - Canada - GetGeoLocation - SOAP SDK");
    console.log("----------------------------------------------------");

    const address = "50 Coach Hill Dr";
    const municipality = "Kitchener";
    const province = "ON";
    const postalCode = "N2E 1P4";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Address        : ${address}`);
    console.log(`Municipality   : ${municipality}`);
    console.log(`Province       : ${province}`);
    console.log(`PostalCode     : ${postalCode}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try {
        const agca = new GetGeoLocationSoap(
            address,
            municipality,
            province,
            postalCode,
            licenseKey,
            isLive,
            timeoutSeconds
        );

        const response = await agca.invokeAsync();

        if (response.Error) {
            console.log("\n* Error *\n");
            console.log(`Error Desc    : ${response.Error.Desc}`);
            console.log(`Error Number  : ${response.Error.Number}`);
            console.log(`Error Location: ${response.Error.Location}`);
        }

        console.log("\n* Geocode Info *\n");

        if (response) {
            console.log(`Latitude  : ${response.Latitude}`);
            console.log(`Longitude : ${response.Longitude}`);
            console.log(`PostalCode: ${response.PostalCode}`);
            console.log(`MatchCode : ${response.MatchCode}`);
        } else {
            console.log("No geocode info found.");
        }
    } catch (e) {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { GetGeoLocationSoapGo };
